src/test/integration/test_scheduler.py

```diff
@@ -0,0 +1,122 @@
+from ja.common.job import JobStatus, Job
+from ja.server.scheduler.default_policies import DefaultCostFunction
+from test.integration.base import IntegrationTest
+from time import sleep
+from typing import Any, List, Tuple
+
+
+class TestScheduler(IntegrationTest):
+    def __init__(self, *args: Any, **kwargs: Any):
+        super().__init__(*args, **kwargs)
+        self.list_updates: List[Tuple[str, JobStatus]] = []
+
+    def _status_updated(self, job: Job) -> None:
+        self.list_updates.append((job.label, job.status))
+
+    def test_three_jobs(self) -> None:
+        """
+        A test where 3 jobs with different priorities are scheduled, and then a worker is added.
+        """
+        self._clients[0].run(
+            self.get_arg_list_add(num_seconds=1, label="lo", priority="0", threads=4, memory=16 * 1024))
+        self._clients[1].run(
+            self.get_arg_list_add(num_seconds=1, label="me", priority="1", threads=4, memory=16 * 1024))
+        self._clients[2].run(
+            self.get_arg_list_add(num_seconds=1, label="hi", priority="3", threads=4, memory=16 * 1024))
+        self._server._database.set_job_status_callback(self._status_updated)
+        self._add_worker(0)
+        sleep(8)
+        expect: List[Tuple[str, JobStatus]] = [
+            ("hi", JobStatus.RUNNING),
+            ("hi", JobStatus.DONE),
+            ("me", JobStatus.RUNNING),
+            ("me", JobStatus.DONE),
+            ("lo", JobStatus.RUNNING),
+            ("lo", JobStatus.DONE),
+        ]
+        self.assertListEqual(self.list_updates, expect)
+
+    def test_parallel_jobs(self) -> None:
+        """
+        A test where 3 jobs are scheduled on 3 workers
+        """
+        self._add_worker(0)
+        self._add_worker(1)
```

Comment:
Aren't there just 2 workers?

Reply:
Oh yes, there used to be 3 but I changed it so that we can also test multiple jobs on the same machine. Let me fix that